### PR TITLE
[Infra Views Read/Write]  Edit/Update Data Bag item

### PR DESCRIPTION
**Pull request by vinay033**
### :nut_and_bolt: Description: What code changed, and why?
Need to add Edit Data Bag Items UI. Only Json is to be edited

### :chains: Related Resources
https://github.com/chef/automate/issues/4542
### :+1: Definition of Done
I have added UI changes to update the data bag item.
### :athletic_shoe: How to Build and Test the Change
```
STEP 1
inside the hab studio

[default:/src:0]# build components/automate-ui-devproxy/
[default:/src:0]# start_automate_ui_background
[default:/src:0]# start_all_services

STEP 2
open new window
go to automate UI path

$ cd components/automate-ui
and run the command 

npm run serve:hab

navigate to `https://a2-dev.test/infrastructure/client-runs`
add some sample data and then test the changes using tab change
```
To add data https://github.com/chef/automate/blob/master/dev-docs/adding-data/adding_test_data.md#adding-data-to-infra-views
```

Go to the `infrastructure tab` from the top navigation bar then you can see the `Chef Server`
If you have data then you can see the list of servers --> then click to any of the server --> see the list of orgs 
--> click to any of org ---> you can see the multiple tabs
--> click to `databags` tab
--> see the list of databags ---> click to any of databag --> you are on databag details page 
--> if databag items available you can click to any of databag 
--> underneath of header you can see the edit button, when you click a modal will be open and you can update the databag item.
```
### :white_check_mark: Checklist

- [ ] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [ ] Tests added/updated?
- [ ] Docs added/updated?
- [ ] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### :camera: Screenshots, if applicable

https://user-images.githubusercontent.com/12297653/111503406-bce0d100-876c-11eb-9cb4-072fad3202ae.mp4

### CODE REVIEW

File: components/automate-ui/src/app/entities/data-bags/data-bag-details.effects.ts [hunk 1]
```diff
@@ -43,4 +46,31 @@ export class DataBagItemsEffects {
...
+        map((resp: DataBagItems) => new UpdateDataBagItemSuccess(resp)),
+        catchError((error: HttpErrorResponse) =>
+          observableOf(new UpdateDataBagItemFailure(error))))));
+
+  @Effect()
+  updateDataBagItemSuccess$ = this.actions$.pipe(
+    ofType(DataBagItemsActionTypes.UPDATE_SUCCESS),
+    map(({ payload: dataBagItem }: UpdateDataBagItemSuccess) => new CreateNotification({
+    type: Type.info,
+    message: `Successfully Updated Data Bag Item - ${dataBagItem.name}.`
```
Comment (msorens) [inline components/automate-ui/src/app/entities/data-bags/data-bag-details.effects.ts:63]: Let's be consistent with the majority of the code base
```suggestion
    message: `Updated data bag item ${dataBagItem.name}.`
```

File: components/automate-ui/src/app/modules/infra-proxy/edit-data-bag-item-modal/edit-data-bag-item-modal.component.html [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+          </div>
+        </chef-form-field>
+      </div>
+      <div id="button-bar">
+        <chef-button tertiary (click)="closeCreateModal()">Cancel</chef-button>
+        <chef-button primary [disabled]="!updateForm?.valid || conflictError || itemAttrParseError || !updateForm.dirty"
+          id="edit-button" (click)="updateDataBagItem()">
+          <chef-loading-spinner *ngIf="updating"></chef-loading-spinner>
+          <span *ngIf="!updating">Save Item</span>
+          <span *ngIf="updating">saving...</span>
```
Comment (msorens) [inline components/automate-ui/src/app/modules/infra-proxy/edit-data-bag-item-modal/edit-data-bag-item-modal.component.html:25]:
```suggestion
          <span *ngIf="updating">Saving...</span>
```

File: components/automate-ui/src/app/modules/infra-proxy/edit-data-bag-item-modal/edit-data-bag-item-modal.component.ts [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+        this.closeCreateModal();
+      });
+  }
+
+  ngOnDestroy(): void {
+    this.isDestroyed.next(true);
+    this.isDestroyed.complete();
+  }
+
+  public handleInput(event: KeyboardEvent): void {
```
Comment (msorens) [inline components/automate-ui/src/app/modules/infra-proxy/edit-data-bag-item-modal/edit-data-bag-item-modal.component.ts:79]: Consistency with other public methods in this file:
```suggestion
  handleInput(event: KeyboardEvent): void {
```

File: components/automate-ui/src/app/entities/data-bags/data-bag-details.effects.ts (hunk 1 above)
Comment (vinay033) [reply components/automate-ui/src/app/entities/data-bags/data-bag-details.effects.ts:63]: @msorens I have incorporated review comment changes, please have a look.

**Comment by msorens** [comment]
Your instructions for navigating in the browser are thorough... except you need to explain how to get data in the first place.

**Comment by vivek-yadav** [comment]
Please put in all the steps to build and run tests in the PR description, and if this is dependent on some other PR to be merged in Master then After that is merged in master please Rebase this branch with master.

**Review by vivek-yadav — APPROVED** [review]
Approved, UI looks fine. edit and update working fine on databag.

**Review by msorens — APPROVED** [review]

**Comment by chaitali-mane** [comment]
While acceptance testing, observed
Edit DataBag Modal width disturb other modal width
will cover in [#4873 ](https://github.com/chef/automate/issues/4873)

**Pull request merged**